pkgs/tools/security/john/default.nix

```
@@ -131,7 +133,7 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "John the Ripper password cracker";
-    license = licenses.gpl2Plus;
+    license = [ licenses.gpl2Plus ] ++ lib.optionals enableUnfree [ licenses.unfree ];
```

Nit: We generally consider UnRAR to be `unfreeRedistributable`.

---

Ah, OK. Does that mean `enableUnfree` can be `true` by default here?

I based this on `7zz` which uses `unfree` for some RAR decompression code, should it be changed there too?

---

```suggestion
    license = [ licenses.gpl2Plus ] ++ lib.optionals enableUnfree [ licenses.unfreeRedistributable ];
```
This is the suggestion, no other changes needed.
Though it'd probably make our lives easier to have a proper unrar license entry.

---

No, it can’t be default, or at least shouldn’t be. Besides ideological reasons, Hydra won’t build even `unfreeRedistributable`.

---

Just sent #328698 to add the unrar license.
Also there is an open tracking issue to possibly build redistributable packages (#83884)